Sources/Confidence/Confidence.swift

```diff
@@ -1,23 +1,40 @@
 import Foundation
 
 public class Confidence: ConfidenceEventSender {
-    public var context: ConfidenceStruct
+    private let parent: ConfidenceEventSender?
```

Comment:
TODO: Carefully verify cases of retain cycles

Reply:
Since the parent will not keep any reference of its children, I think retain cycles are not a problem in this implementation. But would love to have another pair of eyes on this one

Reply:
why do we expect the parent to be type eventsender? in android for example, we have a protocol called `ConfidenceContextProvider` and it has a function that returns a context as `getContext`. `Confidence` will also implements this interface and `self` can be sent as parent to the child still. wdyt?

Reply:
what I mean is the only reason we need to pass a parent to the child here is to be able to read the context from it, so why not just expose it as `ConfidenceContextProvider`. it will be also easier in tests to mock. since we only need to fake one function.